presto-hive/src/test/java/com/facebook/presto/hive/TestHiveBucketing.java

```diff
@@ -289,6 +289,7 @@ public static void appendToBlockBuilder(Type type, Object element, BlockBuilder
             case StandardTypes.BOOLEAN:
                 type.writeBoolean(blockBuilder, (Boolean) element);
                 break;
+            case StandardTypes.INTEGER:
```

Comment:
type doesn't have `writeInteger`, so you have to pass in a long.  This is how the `Date` type works below.